doc/source/serve/deployment-graph.md

```diff
@@ -0,0 +1,545 @@
+
+DAGNode can also be passed into other DAGNode in dag binding. In the full example below, ```Combiner``` calls into two instantiations of ```Model``` class, which can be bound and passed into ```Combiner```'s constructor as if we're passing in two regular python class instances.
+
+```python
+m1 = Model.bind(1)
+m2 = Model.bind(2)
+combiner = Combiner.bind(m1, m2)
+```
+
+Similarly, we can also pass and bind upstream DAGNode results that will be resolved upon runtime to downstream DAGNodes, in our example, a function `run()` that access class method of ```Combiner``` class takes two preprocessing DAGNodes' output as well as part of user input that will be resolved when upstream `DAGNode` are executed.
+
+```python
+preprocessed_1 = preprocessor.bind(dag_input[0])
+preprocessed_2 = avg_preprocessor.bind(dag_input[1])
+...
+dag = combiner.run.bind(preprocessed_1, preprocessed_2, dag_input[2])
+```
+
+```{tip}
+At runtime, calls of deployment node like ` self.m1.forward.remote()` will be automatically replaced with a `.remote()` call to the deployment handle to `self.m1`.
+```
+
++++
+
+### Step 3: Dynamic aggregation based on user input
+
+Now we have the backbone of our DAG setup: splitting and preprocessing user inputs, aggregate into new request data and send to multiple models downstream. Let's add a bit more dynamic flavor in it to demostrate deployment graph is fully python programmable by introducing control flow based on user input.
+
+It's as simple as adding a plain `if / else` on `combiner.run()`, and from our previous binding, we get `operation` field at runtime from user provided input data.
+
+```python
+dag = combiner.run.bind(preprocessed_1, preprocessed_2, dag_input[2])
+```
+
++++
+
+```python
+@serve.deployment
+class Combiner:
+    ...
+
+    async def run(self, req_part_1, req_part_2, operation):
+        # Merge model input from two preprocessors  
+        req = f"({req_part_1} + {req_part_2})"
+        
+        # Submit to both m1 and m2 with same req data in parallel
+        r1_ref = self.m1.forward.remote(req)
+        r2_ref = self.m2.forward.remote(req)
+        
+        # Async gathering of model forward results for same request data
+        rst = await asyncio.gather(*[r1_ref, r2_ref])
+        
+        # Control flow that determines runtime behavior based on user input
+        if operation == "sum":
+            return f"sum({rst})"
+        else:
+            return f"max({rst})"
+```
+
++++
+
+### Step 4: Driver deployment to handle http ingress
+
+Now we've built the entire serve DAG with the topology, args binding and user input. It's time to add the last piece for serve -- a Driver deployment to expose and configure http. We can configure it to start with two replicas in case the ingress of deployment becomes bottleneck of the DAG.
+
+```{note}
+We expect each DAG has a driver class implementation as root, similar to the example below. This is where HTTP ingress are configured and implemented. 
+```
```

Comment:
can you please mention the default http driver? 
and mention this is for advanced only. 
and open an issue for how it currently cannot be support.

Reply:
sg i think we have the issue there already that you're working on, let me add the part for a default DAGDriver.

Reply:
my only concern is unless we have your fix cherrypicked this notebook won't run with default DAGDriver :/

Reply:
we can keep this, just need to rework once default driver is fixed.

Reply:
addressed in latest commit -- please let me know if you think the content looks good !

Reply:
looks good for now!